test/coverage/BUILD

```diff
@@ -0,0 +1,115 @@
+# This file is generated by test/coverage/gen_build.sh. DO NOT EDIT. Rerun
```

Comment:
This feels extremely fragile and nearly guaranteed to fall out of date. Is there any way that test libraries can somehow register themselves to be part of an optional monolithic test process. Furthermore, is there any reason for this even to be coverage specific? I.e., can we just have an option to build a monolithic test binary regardless of whether we are doing ASAN, coverage, etc.? This seems like potentially the way to go in CI where as I've mentioned elsewhere, the extra links don't seem to provide much value?